redbot/core/commands/converter.py

```diff
@@ -162,140 +173,256 @@ class APIToken(discord.ext.commands.Converter):
+            parser understands: (``weeks``, ``days``, ``hours``, ``minutes``, ``seconds``)
+        default_unit : Optional[str]
+            If provided, it will additionally try to match integer-only input into
+            a timedelta, using the unit specified. Same units as in ``allowed_units``
+            apply.
+
+        Returns
+        -------
+        type
+            The converter class, which will be a subclass of `TimedeltaConverter`
+        """
+
+        class PartialMeta(type):
+            __call__ = functools.partialmethod(
+                type(DictConverter).__call__,
+                allowed_units=allowed_units,
+                default_unit=default_unit,
+                minimum=minimum,
+                maximum=maximum,
             )
-        if delta is not None:
-            return delta
-        raise BadArgument()  # This allows this to be a required argument.
 
+        class ValidatedConverter(TimedeltaConverter, metaclass=PartialMeta):
+            pass
 
-def get_timedelta_converter(
-    *,
-    default_unit: Optional[str] = None,
-    maximum: Optional[timedelta] = None,
-    minimum: Optional[timedelta] = None,
-    allowed_units: Optional[List[str]] = None,
-) -> type:
-    """
-    This creates a type suitable for typechecking which works with discord.py's
-    commands.
-    
-    See `parse_timedelta` for more information about how this functions.
+        return ValidatedConverter
 
-    Parameters
-    ----------
-    maximum : Optional[timedelta]
-        If provided, any parsed value higher than this will raise an exception
-    minimum : Optional[timedelta]
-        If provided, any parsed value lower than this will raise an exception
-    allowed_units : Optional[List[str]]
-        If provided, you can constrain a user to expressing the amount of time
-        in specific units. The units you can chose to provide are the same as the
-        parser understands: `weeks` `days` `hours` `minutes` `seconds`
-    default_unit : Optional[str]
-        If provided, it will additionally try to match integer-only input into
-        a timedelta, using the unit specified. Same units as in `allowed_units`
-        apply.
 
-    Returns
-    -------
-    type
-        The converter class, which will be a subclass of `TimedeltaConverter`
-    """
+if TYPE_CHECKING:
+    RedOptional = Optional
+else:
+
+    class RedOptional:
+        """
+        This can be used instead of typing.Optional
+        To avoid discord.py special casing the conversion behavior
+
+        (see also: the ``ignore_optional_for_conversion`` option of commands)
 
-    class PartialMeta(type(TimedeltaConverter)):
-        __call__ = functools.partialmethod(
-            type(DictConverter).__call__,
-            allowed_units=allowed_units,
-            default_unit=default_unit,
-            minimum=minimum,
-            maximum=maximum,
-        )
+        This converter class is still provisional
```

Comment:
This suggestion and the one above can only be applied together as I changed the order.
Also, if `ignore_optional_for_conversion` gets documented, this should be a proper reference to that option.
```suggestion
        .. seealso::
            The ``ignore_optional_for_conversion`` option of commands.
```